pubsub/system-test/topics.test.js

```diff
@@ -13,65 +13,115 @@
+const fullTopicName = `projects/${projectId}/topics/${topicName}`;
+const message = { data: `Hello, world!` };
+const cmd = `node topics.js`;
 
-describe('pubsub:topics', function () {
-  describe('createTopic', function () {
-    it('should create a topic', function (done) {
-      program.createTopic(topicName, function (err, topic, apiResponse) {
-        assert.ifError(err);
-        assert.equal(topic.name, fullTopicName);
-        assert(console.log.calledWith('Created topic: %s', topicName));
-        assert.notEqual(apiResponse, undefined);
-        // Listing is eventually consistent, so give the index time to update
-        setTimeout(done, 5000);
+describe(`pubsub:topics`, () => {
+  after((done) => {
+    pubsub.subscription(subscriptionName).delete(() => {
+      // Ignore any error
+      pubsub.topic(topicName).delete(() => {
+        // Ignore any error
+        done();
       });
     });
   });
 
-  describe('listTopics', function () {
-    it('should list topics', function (done) {
-      program.listTopics(function (err, topics) {
-        assert.ifError(err);
-        assert(Array.isArray(topics));
-        assert(topics.length > 0);
-        var recentlyCreatedTopics = topics.filter(function (topic) {
-          return topic.name === fullTopicName;
-        });
-        assert.equal(recentlyCreatedTopics.length, 1, 'list has newly created topic');
-        assert(console.log.calledWith('Found %d topics!', topics.length));
-        done();
-      });
+  it(`should create a topic`, (done) => {
+    const output = run(`${cmd} create ${topicName}`, cwd);
+    assert.equal(output, `Topic ${fullTopicName} created.`);
+    pubsub.topic(topicName).exists((err, exists) => {
+      assert.ifError(err);
+      assert.equal(exists, true);
+      done();
     });
   });
 
-  describe('publishMessage', function () {
-    it('should publish a message', function (done) {
-      program.publishMessage(topicName, message, function (err, messageIds, apiResponse) {
+  it(`should list topics`, (done) => {
+    // Listing is eventually consistent. Give the indexes time to update.
+    setTimeout(() => {
+      const output = run(`${cmd} list`, cwd);
+      assert.notEqual(output.indexOf(`Topics:`), -1);
+      assert.notEqual(output.indexOf(fullTopicName), -1);
+      done();
+    }, 5000);
+  });
+
+  it(`should publish a simple message`, (done) => {
+    pubsub.topic(topicName).subscribe(subscriptionName, (err, subscription) => {
+      assert.ifError(err);
+      run(`${cmd} publish ${topicName} "${message.data}"`, cwd);
+      subscription.pull((err, messages) => {
         assert.ifError(err);
-        assert(Array.isArray(messageIds));
-        assert(messageIds.length > 0);
-        assert(console.log.calledWith('Published %d message(s)!', messageIds.length));
-        assert.notEqual(apiResponse, undefined);
+        console.log(JSON.stringify(messages, null, 2));
+        assert.equal(messages[0].data, message.data);
```

Comment:
Pub/Sub messages are unordered (they can come in any order).

In this test environment, we know there's only one message because we just created the topic and subscription and published a single message.